### DISCUSSION TITLE
Infered type from schema made everything optional

### DISCUSSION THREAD

**Discussion by PointSingularity** [colinhacks/zod#1170] [Q&A] [answered]
So I was just playing around with zod, and created an empty project just with zod installed.

Here is my only file 

```
// main.ts

import { z } from 'zod';

const dataSchema = z.object({
  firstName: z.string().min(2),
  lastName: z.string().optional(),
  email: z.string().email(),
  phoneNumber: z.string().min(10).optional(),
});

type dataType = z.infer<typeof dataSchema>;

const data: dataType = {};

dataSchema.parse(data);
```
![image](https://user-images.githubusercontent.com/56246797/170460424-281d8325-a3c3-4b27-8698-53f4caa9c234.png)

But as you can see on the image the type is inferred with all properties being optional.
On the other hand, if I execute the code I do get an error:

```
TSError: ⨯ Unable to compile TypeScript:
main.ts:12:7 - error TS2739: Type '{}' is missing the following properties from type '{ lastName?: string | undefined; phoneNumber?: string | undefined; firstName: string; email: string; }': firstName, email

12 const data: dataType = {};
```

And here is the complete stack trace from the tsc compiled js file:
<details>
  <summary>Stack Trace</summary>

D:\wd2\zod-test\node_modules\zod\lib\types.js:123
        throw result.error;
        ^

ZodError: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "firstName"
    ],
    "message": "Required"
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "email"
    ],
    "message": "Required"
  }
]
    at handleResult (D:\wd2\zod-test\node_modules\zod\lib\types.js:28:23)
    at ZodObject.safeParse (D:\wd2\zod-test\node_modules\zod\lib\types.js:140:16)
    at ZodObject.parse (D:\wd2\zod-test\node_modules\zod\lib\types.js:120:29)
    at Object.<anonymous> (D:\wd2\zod-test\main.js:11:12)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Module._load (node:internal/modules/cjs/loader:827:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  issues: [
    {
      code: 'invalid_type',
      expected: 'string',
      received: 'undefined',
      path: [ 'firstName' ],
      message: 'Required'
    },
    {
      code: 'invalid_type',
      expected: 'string',
      received: 'undefined',
      path: [ 'email' ],
      message: 'Required'
    }
  ],
  addIssue: [Function (anonymous)],
  addIssues: [Function (anonymous)]
}

</details>

Any insight would be greatly appreciated.

**Comment by scotttrinh** [comment, marked as answer]
You need to ensure your tsconfig specifies `strict: true`

**Reply by PointSingularity** [reply]
This was it, and it was in the README too.
I need to learn to read more thoroughly 😅

Thank you for the quick answer!

**Reply by scotttrinh** [reply]
Don't feel too bad, we get this same question about once a week! 😅
